webapp/tests/utils/channel_utils.test.jsx

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import * as Utils from 'utils/channel_utils.jsx';
+import TeamStore from 'stores/team_store.jsx';
+import UserStore from 'stores/user_store.jsx';
+import ChannelStore from 'stores/channel_store.jsx';
+import Constants from 'utils/constants.jsx';
+
+describe('Channel Utils', () => {
+    describe('showDeleteOption', () => {
+        test('all users can delete channels on unlicensed instances', () => {
+            global.window.mm_license = {IsLicensed: 'false'};
+            expect(Utils.showDeleteOptionForCurrentUser(null)).
+                toEqual(true);
+        });
+
+        test('users cannot delete default channels', () => {
+            global.window.mm_license = {IsLicensed: 'true'};
+            const channel = {name: Constants.DEFAULT_CHANNEL};
+            expect(Utils.showDeleteOptionForCurrentUser(channel)).
+                toEqual(false);
+        });
+
+        test('system admins can delete private channels, user is system admin test', () => {
+            global.window.mm_license = {IsLicensed: 'true'};
+            global.window.mm_config = {RestrictPrivateChannelDeletion: Constants.PERMISSIONS_SYSTEM_ADMIN};
+            UserStore.isSystemAdminForCurrentUser = () => {
+                return true;
+            };
```

Comment:
Same question as above: I'm trying to create a mock UserStore that always returns true, regardless of the current user. The code under test accesses isSystemAdminForCurrentUser in a static way - it doesn't have an instance of UserStore, so the concern is that if I make isSystemAdminForCurrentUser a mock method in this test, it might still be that mock method in a later test. Or is that not a problem because of some scoping behaviour that I don't know about?